azext_edge/edge/providers/check/base/pod.py

```diff
@@ -114,85 +76,86 @@ def _decorate_pod_condition(condition: bool) -> Tuple[str, str]:
+
+        pod_dict = pod.to_dict()
+        pod_name = pod_dict["metadata"]["name"]
+        pod_phase = pod_dict.get("status", {}).get("phase")
+        pod_conditions = pod_dict.get("status", {}).get("conditions", [])
+        pod_phase_deco, status = decorate_pod_phase(pod_phase)
+
+        check_manager.add_target_eval(
             target_name=target,
             namespace=namespace,
-            display=Padding("- Conditions:", (0, 0, 0, padding)),
+            status=status,
+            resource_name=target_service_pod,
+            value={"name": pod_name, "status.phase": pod_phase},
         )
 
-        for condition in pod_conditions:
-            type = condition.get("type")
-            condition_type = POD_CONDITION_TEXT_MAP[type]
-            condition_status = condition.get("status") == "True"
-            pod_condition_deco, status = _decorate_pod_condition(condition=condition_status)
+        for text in [
+            f"\nPod {{[bright_blue]{pod_name}[/bright_blue]}}",
+            f"- Phase: {pod_phase_deco}"
+        ]:
+            padding = 2 if "\nPod" not in text else 0
+            padding += display_padding
+            check_manager.add_display(
+                target_name=target,
+                namespace=namespace,
+                display=Padding(text, (0, 0, 0, padding)),
+            )
 
-            add_display_and_eval(
-                check_manager=check_manager,
+        # When pod in obnormal state, sometimes the conditions are not available
+        if pod_conditions:
+            check_manager.add_display(
                 target_name=target,
-                display_text=f"{condition_type}: {pod_condition_deco}",
-                eval_status=status,
-                eval_value={"name": pod_name, f"status.conditions.{type.lower()}": condition_status},
-                resource_name=target_service_pod,
                 namespace=namespace,
-                padding=(0, 0, 0, display_padding + 8)
+                display=Padding("- Conditions:", (0, 0, 0, padding)),
             )
 
-            if detail_level > ResourceOutputDetailLevel.summary.value:
-                condition_reason = condition.get("message")
-                condition_reason_text = f"{condition_reason}" if condition_reason else ""
-
-                if condition_reason_text:
-                    # remove the [ and ] to prevent console not printing the text
-                    condition_reason_text = condition_reason_text.replace("[", "\\[")
-                    check_manager.add_display(
-                        target_name=target,
-                        namespace=namespace,
-                        display=Padding(
-                            f"[red]Reason: {condition_reason_text}[/red]",
-                            (0, 0, 0, display_padding + 8),
-                        ),
-                    )
+            for condition in pod_conditions:
+                type = condition.get("type")
```

Comment:
get will return None if "type" key doesn't exist, is that an actual possibility? If not, this can change to `condition["type"]`, otherwise None should be handled.